Compilation Issue on rust 1.41.0

```rust
error[E0428]: the name `imp` is defined multiple times
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/memchr-2.3.1/src/lib.rs:148:5
    |
139 |     fn imp(n1: u8, haystack: &[u8]) -> Option<usize> {
    |     ------------------------------------------------ previous definition of the value `imp` here
...
148 |     fn imp(n1: u8, haystack: &[u8]) -> Option<usize> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `imp` redefined here
    |
    = note: `imp` must be defined only once in the value namespace of this block
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -129,6 +129,27 @@ jobs:
       name: Run benchmarks as tests
       run: cargo bench --manifest-path bench/Cargo.toml --verbose -- --test
 
+  test-with-miri:
+    name: test with miri
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly
+        profile: minimal
+        override: true
+        components: miri
+    - name: Show CPU info for debugging
+      run: lscpu
+    - run: cargo miri test --verbose
+    - run: cargo miri test --verbose --no-default-features
+    - run: cargo miri test --verbose --features libc
+
   rustfmt:
     name: rustfmt
     runs-on: ubuntu-18.04

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -99,9 +99,9 @@ jobs:
       run: |
         set -x
 
-        # Allow using libc while SIMD is available.
+        # Enable libc while using SIMD, libc won't be used.
         # (This is to ensure valid logic in the picking process.)
-        RUSTFLAGS="--cfg memchr_runtime_simd" cargo test --verbose --features libc
+        cargo test --verbose --features libc
 
         preamble="--cfg memchr_disable_auto_simd"
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -98,6 +98,11 @@ jobs:
       name: Run under different SIMD configurations
       run: |
         set -x
+
+        # Allow using libc while SIMD is available.
+        # (This is to ensure valid logic in the picking process.)
+        RUSTFLAGS="--cfg memchr_runtime_simd" cargo test --verbose --features libc
+
         preamble="--cfg memchr_disable_auto_simd"
 
         # Force use of fallback without libc.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -36,7 +36,7 @@ compile_error!("memchr currently not supported on non-32 or non-64 bit");
 #[cfg(feature = "std")]
 extern crate core;
 
-#[cfg(test)]
+#[cfg(all(test, not(miri)))]
 #[macro_use]
 extern crate quickcheck;
 
@@ -51,7 +51,10 @@ mod c;
 mod fallback;
 mod iter;
 mod naive;
-#[cfg(test)]
+#[cfg(all(test, not(miri)))]
+mod tests;
+#[cfg(all(test, miri))]
+#[path = "tests/miri.rs"]
 mod tests;
 #[cfg(all(not(miri), target_arch = "x86_64", memchr_runtime_simd))]
 mod x86;

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -142,7 +142,8 @@ pub fn memchr(needle: u8, haystack: &[u8]) -> Option<usize> {
 
     #[cfg(all(
         memchr_libc,
-        not(all(target_arch = "x86_64", memchr_runtime_simd, miri))
+        not(all(target_arch = "x86_64", memchr_runtime_simd)),
+        not(miri),
     ))]
     #[inline(always)]
     fn imp(n1: u8, haystack: &[u8]) -> Option<usize> {
@@ -312,7 +313,8 @@ pub fn memrchr(needle: u8, haystack: &[u8]) -> Option<usize> {
     #[cfg(all(
         memchr_libc,
         target_os = "linux",
-        not(all(target_arch = "x86_64", memchr_runtime_simd, miri))
+        not(all(target_arch = "x86_64", memchr_runtime_simd)),
+        not(miri)
     ))]
     #[inline(always)]
     fn imp(n1: u8, haystack: &[u8]) -> Option<usize> {

--- a/src/tests/miri.rs
+++ b/src/tests/miri.rs
@@ -1,0 +1,19 @@
+// Simple tests using MIRI
+
+use crate::{memchr, memchr2, memchr3, memrchr, memrchr2, memrchr3};
+
+#[test]
+fn test_with_miri() {
+    assert_eq!(memchr(b'a', b"abcda"), Some(0));
+    assert_eq!(memchr(b'z', b"abcda"), None);
+    assert_eq!(memchr2(b'a', b'z', b"abcda"), Some(0));
+    assert_eq!(memchr2(b'z', b'y', b"abcda"), None);
+    assert_eq!(memchr3(b'a', b'z', b'b', b"abcda"), Some(0));
+    assert_eq!(memchr3(b'z', b'y', b'x', b"abcda"), None);
+    assert_eq!(memrchr(b'a', b"abcda"), Some(4));
+    assert_eq!(memrchr(b'z', b"abcda"), None);
+    assert_eq!(memrchr2(b'a', b'z', b"abcda"), Some(4));
+    assert_eq!(memrchr2(b'z', b'y', b"abcda"), None);
+    assert_eq!(memrchr3(b'a', b'z', b'b', b"abcda"), Some(4));
+    assert_eq!(memrchr3(b'z', b'y', b'x', b"abcda"), None);
+}

